src/plugins/intel_cpu/src/nodes/tensoriterator.cpp

```diff
@@ -351,15 +322,13 @@ bool TensorIterator::isSupportedOperation(const std::shared_ptr<const ov::Node>&
 }
 
 TensorIterator::TensorIterator(const std::shared_ptr<ov::Node>& op, const dnnl::engine& eng, WeightsSharing::Ptr &cache) :
-        Node(op, eng, cache, InternalDynShapeInferFactory()), ngraphOp(op) {
+        Node(op, eng, cache, InternalDynShapeInferFactory()) {
     std::string errorMessage;
     if (!isSupportedOperation(op, errorMessage)) {
         IE_THROW(NotImplemented) << errorMessage;
     }
-}
 
-void TensorIterator::getSupportedDescriptors() {
-    auto tiOp = ov::as_type_ptr<const ov::op::util::SubGraphOp>(ngraphOp);
+    auto tiOp = ov::as_type_ptr<const ov::op::util::SubGraphOp>(op);
```

Comment:
Are we sure that we want to create the CPU subgraph in the node constructor? Usually we want to postpone any heavy operations to stages that happens after the CPU graph transformations are performed, because there might be cases when the node is removed from the graph and all the heavy initializations are not necessary at all. That is why the subgraph is initialized on the `getSupportedDescriptors()` stage.

Reply:
Good point! Thank you for the comment!
As I can see, CPU Plugin graph optimizations don't  work with `Loop` or `TensorIterator` ops so we can initialize class fields in ctor. But I got your point so let's return the changes to avoid heavy initializations in new transformations in the future